src/indicators/drought/ndvi.py

```diff
@@ -0,0 +1,192 @@
+    year = date_obj.year
+    dekad = (
+        (date_obj.timetuple().tm_mday // 10) + ((date_obj.month - 1) * 3) + 1
+    )
+    return year, dekad
+
+
+def _dekad_to_date(year: int, dekad: int):
+    month = ((dekad - 1) // 3) + 1
+    day = 10 * ((dekad - 1) % 3) + 1
+    return datetime(year, month, day)
+
+
+def _fp_date(fp):
+    return _dekad_to_date(2000 + int(fp[2:4]), int(fp[4:6]))
+
+
+def download_ndvi(
+    start_date: Union[date, str, List[int], None] = None,
+    end_date: Union[date, str, List[int], None] = None,
+    redownload: bool = False,
+):
+    if start_date is None:
+        start_year, start_dekad = 2002, 19
+    elif isinstance(start_date, list):
+        start_year, start_dekad = start_date[0], start_date[1]
+    else:
+        start_year, start_dekad = _date_to_dekad(start_date)
+
+    if isinstance(end_date, list):
+        end_year, end_dekad = end_date[0], end_date[1]
+    else:
+        if end_date is None:
+            end_date = date.today()
+        end_year, end_dekad = _date_to_dekad(end_date)
+
+    i_year = start_year
+    i_dekad = start_dekad
+
+    if not redownload:
+        dts = [_fp_date(filename.stem) for filename in _RAW_DIR.glob("*.tif")]
+        if dts:
+            min_year, min_dekad = _date_to_dekad(min(dts))
+            max_year, max_dekad = _date_to_dekad(max(dts))
+            if (start_year < min_year) or (
+                (min_year == start_year) and start_dekad < min_dekad
+            ):
+                if min_dekad == 1:
+                    min_dekad = 36
+                    min_year -= 1
+                else:
+                    min_dekad -= 1
+                download_ndvi(
+                    start_date=[start_year, start_dekad],
+                    end_date=[min_year, min_dekad],
+                    redownload=True,
+                )
+            if (end_year > max_year) or (
+                (max_year == end_year) and end_dekad > max_dekad
+            ):
+                if max_dekad == 36:
+                    max_dekad = 1
+                    max_year += 1
+                else:
+                    max_dekad += 1
+                download_ndvi(
+                    start_date=[max_year, max_dekad],
+                    end_date=[end_year, end_dekad],
+                    redownload=True,
+                )
+    else:
+        while not (
+            i_year > end_year or (i_year == end_year and i_dekad > i_dekad)
+        ):
+            _download_ndvi(year=i_year, dekad=i_dekad)
+            i_dekad += 1
+            if i_dekad > 36:
+                i_year += 1
+                i_dekad = 1
+
+
+def _get_processed_path(iso3):
+    _processed_dir = Path(
+        Path(os.getenv("AA_DATA_DIR"), "public", "processed", iso3, "ndvi")
+    )
+    _processed_dir.mkdir(parents=True, exist_ok=True)
+    return Path(
+        _processed_dir,
+        "{iso3}_ndvi_anomaly.csv".format(iso3=iso3),
+    )
+
+
+def process_ndvi(iso3, geometries, thresholds):
```

Comment:
this does still take quite long to do. Which isn't bad but would add a comment about it. Plus maybe a progress bar could be an idea?

Reply:
Yeah agreed.